src/renderer/components/cluster-manager/lens-views.ts

```diff
@@ -10,21 +11,20 @@ export interface LensView {
 }
 
 export const lensViews = observable.map<ClusterId, LensView>();
+const lensViewsLock = new AwaitLock();
 
 export function hasLoadedView(clusterId: ClusterId): boolean {
   return !!lensViews.get(clusterId)?.isLoaded;
 }
 
 export async function initView(clusterId: ClusterId) {
+  await lensViewsLock.acquireAsync();
+
   if (!clusterId || lensViews.has(clusterId)) {
     return;
```

Comment:
shouldn't the lock be released here before returning? Maybe use try block for all the code in this function, with the release done in finally?
RAII would be nice for releasing the lock :)